### DISCUSSION TITLE
`useLiteralKeys` rule should cover object properties

### DISCUSSION THREAD

**Discussion by Conaclos** [rome/tools#4664] [Suggestions]
The [`useLiteralKeys`](https://docs.rome.tools/lint/rules/useLiteralKeys/) lint rule reports code where static property access can be used instead of computed acces.

```diff
- a["b"]
+ a.b
```

This implements the EsLint [`dot-notation`](https://eslint.org/docs/latest/rules/dot-notation) rule.

The EsLint [`no-useless-computed-key`](https://eslint.org/docs/latest/rules/no-useless-computed-key) rule reports useless computed keys on an object literal:

```diff
{
-  ["b"]: 0,
-  ["1+1"]: 2,
-  ["f"]() {},
+  b: 0,
+  "1+1": 2,
+  f() {},
}
```

I suggest merging these two EsLint rule into [`useLiteralKeys`](https://docs.rome.tools/lint/rules/useLiteralKeys/).
The only minor inconsistency in the rule name is the simplification of `["1+1"]` to `"1+1"`.
The rule could also handle the following cases:

```diff
{
-  "b": 0,
- "f"() {},
+  b: 0,
+ "f"() {},
}
```

```diff
class C {
-  ["b"] = 0
-  "c" = 0
- ["f"]() {}
-  "g"() {}
+  b = 0
+  c = 0
+  f() {}
+  g() {}
}
```

This could also be done on interfaces and object types.

This is not a breaking change since `useLiteralKeys` is a nursery rule.

**Comment by ematipico** [comment]
Great suggestion!
